packages/ragbits-core/src/ragbits/core/vector_stores/chroma.py

```diff
@@ -165,17 +168,77 @@ async def list(
         embeddings = results.get("embeddings") or []
         documents = results.get("documents") or []
         metadatas = (
-            [json.loads(metadata["__metadata"]) for metadata in results.get("metadatas", [])]  # type: ignore
-            if self._metadata_store is None
-            else await self._metadata_store.get(ids)
+            results.get("metadatas") or [] if self._metadata_store is None else await self._metadata_store.get(ids)
         )
 
         return [
             VectorStoreEntry(
                 id=id,
                 key=document,
                 vector=list(embedding),
-                metadata=metadata,  # type: ignore
+                metadata=self._unflatten_dict(metadata) if metadata else {},  # type: ignore
             )
             for id, metadata, embedding, document in zip(ids, metadatas, embeddings, documents, strict=True)
         ]
+
+    @staticmethod
+    def _flatten_dict(d: dict[str, Any], parent_key: str = "", sep: str = ".") -> dict[str, str | int | float | bool]:
```

Comment:
This is such a generic functionality it could be considered an util function.

Reply:
moved to utils